function/avro-import.py

```diff
@@ -0,0 +1,156 @@
+
+    try:
+        job_id = request_json['jobId']
+    except:
+        return handle_exception("error", start_time, "Could not find a jobId in the following request made at " + start_time + ": " + str(request.json))
+
+    try:
+        url = request_json['url']
+        user = request_json['user']
+        user_email = user['userEmail']
+        user_subject_id = user['userSubjectId']
+        workspace = request_json['workspace']
+        workspace_name = workspace['name']
+        workspace_namespace = workspace['namespace']
+    except KeyError as ke:
+        return handle_exception(job_id, start_time, "Key Error: " + str(ke) + " in request " + str(request.json))
+
+    defaults = {'b64-decode-enums': True, 'prefix-object-ids': True}
+    request_options = request_json.get('options', {})
+    options = {**defaults, **request_options}
+
+    try:
+        avro = urllib.request.urlopen(url)
+    except urllib.error.URLError as ue:
+        return handle_exception(job_id, start_time, "URL Error - the url " + url + " is not valid.")
+
+    try:
+        reader = fastavro.reader(avro)
+        translation = translate(reader, options)
+        metadata = {
+            "namespace": workspace_namespace,
+            "name": workspace_name,
+            "userSubjectId": user_subject_id,
+            "userEmail": user_email,
+            "jobId": job_id,
+            "startTime": start_time,
+        }
+        metadata_json_str = str(json.dumps(metadata))
+        metadat_file_name = job_id + "/metadata.json"
+        upsert_json_str = str(json.dumps(translation))
+        upsert_file_name = job_id + "/upsert.json"
+        write_to_bucket(metadat_file_name, metadata_json_str)
+        write_to_bucket(upsert_file_name, upsert_json_str)
+    except Exception as e:
+        return handle_exception(job_id, start_time, "The following exception occurred: " + str(e))
+    except:
+        return handle_exception(job_id, start_time, "Something went wrong with the following request: " + str(request.json))
+
+
+def handle_exception(subdirectory_name, start_time, message):
+    error_file_name = subdirectory_name + "/error-" + start_time + ".txt"
+    write_to_bucket(error_file_name, message)
+    return message
+
+
+def write_upsert_to_bucket(job_id, content_string):
+    compressed_value = gzip.compress(bytes(content_string, 'utf-8'))
+    file_name = job_id + "/upsert.json"
+    write_to_bucket(file_name, compressed_value)
+
+
+def write_metadata_to_bucket(job_id, content_string):
+    file_name = job_id + "/metadata.json"
+    write_to_bucket(file_name, content_string)
+
+
+def write_to_bucket(file_name, content):
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket("avro-translated-json")
+    blob = bucket.blob(file_name)
+    blob.upload_from_string(content)
+
+
+def translate(reader, options=None):
+    if options is None:
+        options = {}
+    t = Translator(options)
+    return t.translate(reader)
```

Comment:
suggestion: scrap this function and just have the caller do `Translator(options).translate(reader)` ?